cudapoa/src/cudapoa_generate_msa.cu

```diff
@@ -180,14 +184,15 @@ __global__ void generateMSAKernel(uint8_t* nodes_d,
                                        node_marks,
                                        check_aligned_nodes,
                                        nodes_to_visit,
-                                       cuda_banded_alignment);
+                                       cuda_banded_alignment,
+                                       (uint16_t)max_nodes_per_window);
```

Comment:
this is going to get tricky. I think we need to check for limits on the size of the types, or convert them all to `uint32_t`. The casting may get problematic (although unlikely in this case) if 65k+ number gets rolled over to uint16_t

Reply:
I agree, this is not safe for cases where number of nodes per window exceeds INT16_MAX = 32767. However, if we want to make it template, we got to template all data structures in cudapoa_kernels.cuh (WindowDetails, InputDetails, GraphDetails...).